doc/api/assert.md

```diff
@@ -1063,6 +1063,31 @@ If the values are not strictly equal, an `AssertionError` is thrown with a
 `message` parameter is an instance of an [`Error`][] then it will be thrown
 instead of the `AssertionError`.
 
+## assert.stringMatching(input, match)
+<!-- YAML
+added: v10.11.1
+-->
+* `input` {string|string[]}
+* `match` {RegExp|RegExp[]|string|string[]}
+
+Verify that requested constraints match the provided input :
+
+```js
+const assert = require('assert');
+
+// default case : check if string(s) match criterias
+// should be false
+assert.stringMatching('blublu', ['blublu', 'blu']);
+
+// advanced case : check that strings matches regex
+// should be true
+assert.stringMatching(['blublu', 'blu'], /blu\.*/);
+
+// or to mix both of them ^^
+// should be false
+assert.stringMatching(['blublu', 'lol'], ['blublu', /blu\.*/]);
```

Comment:
How do I interpret this? Are both strings required to be equal to `'blublu'` *and* to match `/blu\.*/`?

Reply:
Yes but maybe this example is not so clear .
To make a better example , I should change the 'blublu' (the one in the match parameter) to a second regular expression just to show the custom possibilities